src/main/java/com/rocketden/main/exception/api/ApiExceptionHandler.java

```diff
@@ -14,4 +19,19 @@
         ApiError apiError = e.getError();
         return new ResponseEntity<>(apiError.getResponse(), apiError.getStatus());
     }
+
+    // Handle failure to convert from string to enum (e.g. "easy" -> Difficulty.EASY)
+    @Override
+    protected ResponseEntity<Object> handleHttpMessageNotReadable(HttpMessageNotReadableException ex,
+                                                                  HttpHeaders headers, HttpStatus status,
+                                                                  WebRequest request) {
+
+        // Check for a RoomError.BAD_SETTING triggered in the Difficulty.fromString method
+        if (ex.getCause() != null && ex.getCause().getCause() instanceof ApiException) {
+            ApiError apiError = RoomError.BAD_SETTING;
+            return new ResponseEntity<>(apiError.getResponse(), apiError.getStatus());
+        }
+
+        return super.handleHttpMessageNotReadable(ex, headers, status, request);
```

Comment:
I could be wrong but I think it also results anytime when your request body content is empty, as in the following:

<img width="943" alt="Screenshot" src="https://user-images.githubusercontent.com/24768574/95609328-6039f700-0a13-11eb-8a22-7c0b936bffa2.png">

There might be other cases as well, but I'm not too sure. If you think it's good I can also make a custom error for this situation too like EMPTY_CONTENT or something (although idk if that might mess up some other edge cases not yet discovered)

Reply:
I'm just thinking of how the JavaScript will handle this situation. If a bug occurs where the body content is empty due to some error of not being sent properly, we'll receive nothing. I can easily see this causing an error on the frontend, as I believe this would go into the `then` response, which assumes that the result is of a certain format.

I think `EMPTY_CONTENT` might be necessary because of this -- but let me know what you think. Should we handle the case above, and if so, how should we do that?

Reply:
Actually, scratch that -- I realized that anytime we have just `{}` at the very least, it will still return `INVALID_PERMISSIONS`. I'm free to have a conversation on this, but I don't think this is an edge case we have to account for.